pages/index.tsx

```diff
@@ -20,35 +20,13 @@ export const getServerSideProps = async () => {
   const client = getApolloClient();
   const randomPokemonId = getRandomPokemonId();
 
-  const { data } = await client.query<Pokemon>({
-    query: gql`
-      query FeaturedPokemon($id: String!) {
-        pokemon(id: $id) {
-          id
-          number
-          name
-          weight {
-            minimum
-            maximum
-          }
-          height {
-            minimum
-            maximum
-          }
-          classification
-          types
-          resistant
-          weaknesses
-          fleeRate
-          maxCP
-          maxHP
-          image
-        }
-      }
-    `,
-    variables: {
-      id: randomPokemonId,
-    },
+  const variables: PokemonQueryVariables = {
+    id: randomPokemonId,
+  };
+
+  const { data } = await client.query<PokemonQuery>({
+    query: PokemonDocument,
+    variables,
```

Comment:
All of the other code now gets reduced to these 3 lines of code. These will now fully type the return value, and ensure we always pass a valid query/variables to the API.